### DISCUSSION TITLE
Is there a ~`type Id<T> = unknown & { [P in keyof T]: T[P] }` in this lib?

### DISCUSSION THREAD

**Discussion by ftzi** [millsp/ts-toolbelt#264] [Q&A] [answered]
This is a common type I use in my codes for some time, that once I found in a SO answer.

It basically just turn objects types into their raw forms, so the properties are readable on hover, and not just the type name.

After a few years writing those types by hand or copying from other codes of mine, I am finally tired of doing so, and I am finally willing to use a common lib for those general type stuff so I can improve my productivity.

Does this lib has this functionality?

**Comment by tonivj5** [comment, marked as answer]
I think you're looking for `Compute` https://github.com/millsp/ts-toolbelt/blob/master/sources/Any/Compute.ts

**Reply by ftzi** [reply]
Exactly this! Many thanks for the instant answer! lol

**Reply by Ayub-Begimkulov** [reply]
Could you explain what is a point of the `& unknown`?
Is it used here to force TS to compute the type (similar to using `& {}` in mapped types like `OptionalFlat`)?
But it seems like it already works without an intersection. 
Is there an example where it wont work without `& unknonw`?

**Reply by ftzi** [reply]
I had the same question months ago and I removed the unknown from it. But, there were specific cases that the resulting type was bad, then I re-added it and they were fixed.

I can't really remember what cases it served, but what I have as conclusion is that it's indeed necessary.
